src/utils/Home.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { addCoin } from '../redux/coin/coin';
+
+const Home = () => {
+  const [query, setQuery] = useState('');
+  const dispatch = useDispatch();
+
+  const isLoading = useSelector((state) => state.coins.loading);
+  const allCoins = useSelector((state) => state.coins.data.coins);
+
+  return (
+    <div data-testid="home-1" className="">
+      <div className="w-[100%] flex justify-center items-center">
+        <form className="text-white flex items-center justify-center w-[90%] md:w-[60%] m-2">
+          <input onKeyUp={(e) => { setQuery(e.target.value.toLowerCase()); }} className="bg-[#fc5193] px-4 py-2 outline-none w-[80%]  focus:border-violet-600/50 focus:shadow-sm shadow-violet-400" type="text" placeholder="Search Coin name" required />
+          <input onClick={(e) => { e.preventDefault(); setQuery(query); }} className="py-2 px-4 bg-[#fc5193] border hover:bg-violet-400/75 sm:w-[20%] cursor-pointer" type="button" value="Search" />
+        </form>
+      </div>
+      <div>
+        <ul className="grid grid-cols-2 [&>*:nth-child(4n+2)]:bg-[#d1447a] [&>*:nth-child(4n-1)]:bg-[#d1447a]">
+          {allCoins?.filter((coin) => (coin.name.toLowerCase().includes(query)))
+            .map((coinf) => (
+              <Link to="/coin" onClick={() => { dispatch(addCoin(coinf)); }} key={coinf.id}>
+                <li className=" flex flex-col text-right justify-center py-2 px-4 m-2">
+                <i className="fa-solid fa-circle-arrow-right ml-auto" />
+                  <img className="w-20 h-20 ml-auto mr-auto" src={coinf.icon} alt={coinf.id} />
+                  <h2 className="md:text-xl text-lg font-semibold">{coinf.name}</h2>
+                  <span className={coinf.priceChange1d < 0 ? 'text-white text-sm ml-auto' : 'text-white text-sm ml-auto'}>
+                    {coinf.priceChange1d}
+                    {' '}
+                    <i className={coinf.priceChange1d < 0 ? 'fa-solid fa-arrow-trend-down' : 'fa-solid fa-arrow-trend-up'} />
+                  </span>
+                </li>
+              </Link>
+            ))}
+        </ul>
```

Comment:
- [ ] Please also work on your markup structure here, as of now, this code would produce something like this:

```js
<ul>
  <a>
    <li> ... </li>
  </a>
  <a>
    <li> ... </li>
  </a>
...
</ul>

```

This markup structure is invalid due to the child of `<ul>` not being `<li>` list items, wouldn't it be better to end up with something like this?: 🤖 


```js
<ul>
  <li>
    <a> ... </a>
  </li>
  <li>
    <a> ... </a>
  </li>
...
</ul>

```

Please consider applying this change! you can read [more here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li). 🤓 🔧